agent/config/builder.go

```diff
@@ -1852,10 +1845,18 @@ func (b *Builder) stringValWithDefault(v *string, defaultVal string) string {
 	return *v
 }
 
+// Deprecated: use stringVal
```

Comment:
This is a bit confusing unless you look down a bit, which wasn't my first intention. It's a silly nit, but this might benefit from being reworded to have something like "Please look a couple lines down" perhaps?

Reply:
Ah yes, because both the Builder method and the function are named the same thing, that is confusing! I'll reword this to clarify.